src/utils/string.js

```diff
@@ -126,32 +131,61 @@ export const stringsToMarkerRegExps = (
     return null;
   }
 
-  const fieldStrings = new Map();
-  const genericStrings = [];
+  // We create this map to group all the field specific search strings and then
+  // we aggregate them to create a single regexp for each field later.
+  const fieldStrings: Map<
+    string,
+    {| positive: string[], negative: string[] |},
+  > = new Map();
+  // These are the non-field specific search strings. They have to be positive
+  // as we don't support negative generic filtering.
+  const genericPositiveStrings = [];
   for (const string of strings) {
-    const prefixMatch = string.match(/^([a-z0-1]+):(.+)/i);
+    // First capture group is used to determine if it has a "-" in front of the
+    // field to understand if it's a negative filter.
+    // Second capture group is used to get the field name.
+    // Third capture group is to get the filter value.
+    const prefixMatch = string.match(/^(-*)([a-z0-1]+):(.+)/i);
     if (prefixMatch) {
```

Comment:
It could be easier to follow if you name the matches in individual variables.
It could also be a good idea to use the new regexp capability of named capturing groups (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group) as it seems to be supported by everyone for some time now. Then you could do something such as:
```js
const prefixMatch = string.match(/^(?<maybeNegative>-?)(?<key>\w+):(?<value>.+)/i);
if (prefixMatch) {
  const { maybeNegative, key, value } = prefixMatch.groups;
  ...
}
```
(with the changes I outline in my other comments)